src/api/extendobservable.ts

```diff
@@ -65,10 +65,19 @@ export function extendObservable<A extends Object, B extends Object>(
     try {
         const decorators: any = {}
         for (let key in properties) {
-            // TODO: using decorators is a bit inefficient, short circuit those
-            if ((target as any) === properties && !isPropertyConfigurable(target, key)) continue // see #111, skip non-configurable or non-writable props for `observable(object)`.
             const descriptor = Object.getOwnPropertyDescriptor(properties, key)!
-            if (typeof descriptor.get === "function") {
+            const { value, get } = descriptor
+            if (process.env.NODE_ENV !== "production") {
+                if (Object.getOwnPropertyDescriptor(target, key))
+                    fail(
+                        `'extendObservable' can only be used to introduce new properties. Use 'decorate' to update existing properties. The property '${key}' already exists on '${target}'`
+                    )
+                if (isComputed(value))
```

Comment:
It shouldn't throw, but simply assign boxed computed:
```javascript
const o = {}
mobx.extendObservable(o, {
  boxedComputed: mobx.computed(() => "a"), 
});
console.log(o.boxedComputed.get());
```

Reply:
This would create an observable references to a computed object, it is barely imaginable that that is what the user would intend, so throwing and telling how to properly do it is better imho. 

If he really needs a boxed computed, he can just do `o.boxedComputed = mobx.computed()`. Or in the weird case hey needs an observable ref to it, extending it with `null` first, and then assigning it would work fine.